### PR TITLE
Fix js_template method redefined warning from Rails tests

### PR DESCRIPTION
**Pull request by prathamesh-sonpatki**

### CODE REVIEW

**Comment by prathamesh-sonpatki** [comment]
r? @rafaelfranca 

Ran locally and verified that there is no warning.

```
$ be ruby -w -Itest:lib:/../activesupport/lib:/../actionpack/lib:/../activemodel/lib test/application/configuration/custom_test.rb
Run options: --seed 33889

# Running:

.

Finished in 0.961010s, 1.0406 runs/s, 6.2434 assertions/s.

1 runs, 6 assertions, 0 failures, 0 errors, 0 skips
```

File: lib/coffee/rails/js_hook.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ module JsHook
       extend ActiveSupport::Concern
 
       included do
+        remove_method :js_template
```
Comment (kaspth) [inline lib/coffee/rails/js_hook.rb]: How does this work with Rails 4.2 that doesn't have this method?

Should we opt for `remove_possible_method` or `redefine_method`?

Comment (kaspth) [inline lib/coffee/rails/js_hook.rb]: Looked at the build failure. Yea, we should switch to one of these methods (I personally prefer the latter).

Comment (prathamesh-sonpatki) [inline lib/coffee/rails/js_hook.rb]: :+1: Switched to `redefine_method`. The only side-effect is Rails < 5 will also get `js_template` method now which will be unused.

**Comment by prathamesh-sonpatki** [comment]
Build is green.

**Comment by kaspth** [comment]
Neat!

**Pull request merged**